src/connectionhealthpolicy/SendingAudioFailureConnectionHealthPolicy.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import BaseConnectionHealthPolicy from './BaseConnectionHealthPolicy';
+import ConnectionHealthData from './ConnectionHealthData';
+import ConnectionHealthPolicy from './ConnectionHealthPolicy';
+import ConnectionHealthPolicyConfiguration from './ConnectionHealthPolicyConfiguration';
+
+export default class SendingAudioFailureConnectionHealthPolicy
+  extends BaseConnectionHealthPolicy
+  implements ConnectionHealthPolicy {
+  private readonly secondsToConsider: number;
+  private readonly initialWaitTimeMs: number;
+  private readonly coolDownTimeMs: number;
+  private readonly maximumTimesToWarn: number;
+  private warnCount: number;
+  private lastWarnTimestampMs: number;
+
+  constructor(
+    private logger: Logger,
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData
+  ) {
+    super(configuration, data, 'Sending Audio Health');
+    this.secondsToConsider = configuration.sendingAudioFailureSamplesToConsider;
+    this.initialWaitTimeMs = configuration.sendingAudioFailureInitialWaitTimeMs;
+    this.maximumTimesToWarn = configuration.maximumTimesToWarn;
+    this.coolDownTimeMs = configuration.cooldownTimeMs;
+    this.lastWarnTimestampMs = 0;
+    this.warnCount = 0;
+  }
+
+  private isSendingAudioUnHealthy(): boolean {
+    const hasEnoughTimeElapsedToEvaluateStatus = !this.currentData.isConnectionStartRecent(
+      this.initialWaitTimeMs
+    );
+    const areAudioPacketsNotBeingSent =
+      this.currentData.consecutiveStatsWithNoAudioPacketsSent >= this.secondsToConsider;
+    return hasEnoughTimeElapsedToEvaluateStatus && areAudioPacketsNotBeingSent;
+  }
+
+  health(): number {
+    const didWarnRecently = Date.now() - this.lastWarnTimestampMs < this.coolDownTimeMs;
+    if (this.isSendingAudioUnHealthy()) {
+      if (this.currentHealth > this.minimumHealth() && !didWarnRecently) {
+        this.logger.debug(`Sending Audio is unhealthy for ${this.secondsToConsider} seconds`);
+        this.warnCount++;
+        if (this.warnCount > this.maximumTimesToWarn) {
+          this.logger.debug(
+            'SendingAudioFailure maximum warnings were breached. Falling back to reporting healthy.'
+          );
+          return this.maximumHealth();
```

Comment:
So why not `currentHealth`, technically, the experience would be still bad right?

Reply:
Yeah, the experience would still be bad but this is how `UnusableAudioWarningConnectionHealthPolicy` also works. I think the rationale with returning maximum health is that we don't want to flood the observer with too many invocations which might directly lead to user level warnings on UI if builders don't have any extra handling in between.

I don't think an ignore flag makes sense here, because `maximumTimesToWarn` itself is a configurable value. Instead maybe I can create a separate `sendingAudioFailureMaximumTimesToWarn` which has a higher value (say 5 instead of 2)? Let me know your thoughts.

Reply:
I think using `maximumTimesToWarn` is fine for now and we can add extra variable if needed later. 

If ignoring then why would we flood the observers? Instead of checking for `maximumHealth` value will have to also check whether `health()` returns ignore. But anyways if other policies are using the same I am fine with keeping it that way.

Reply:
We call the `healthIfChanged()` method for each health policy which returns the health value if the health has changed, null if the health hasn't changed so by ignoring the unhealthy state and returning `maximumHealth()` there would be no change and hence observers won't be notified.